File: Semester 1/Class02_Computer_Contest_Lv_2_Notes_and_Homework_Sep_18/lvl2_hw_sol/lvl2_secret_instructions.java
import java.util.*;
import java.io.*;
public class test2 {
	public static void main(String[] args) throws IOException{
		Scanner in = new Scanner(System.in);
		String dir = "", pre = "";
		while(true) {
			int n = in.nextInt(); 
			if(n == 99999) break;
			int a = n/10000, b = n/1000%10, c = n%1000;
			if((a+b)%2 != 0) dir = "left";
			else if(a + b == 0) dir = pre;
			else dir = "right";
			pre = dir;
			System.out.println(dir + " " + c);
		}
	}
}

File: Semester 1/Class02_Computer_Contest_Lv_2_Notes_and_Homework_Sep_18/lvl2_hw_sol/lvl2_secret_instructions.cpp
#include <bits/stdc++.h>
using namespace std;
#define f first
#define s second
#define lc (rt<<1)
#define rc (rt<<1|1)
#define pb push_back
#define cl(a, b) memset(a, b, sizeof(a))
#define mp(a, b) make_pair((a), (b))
#define all(x) x.begin(),x.end()
typedef long long ll;
typedef pair<int, int> pi;
typedef pair<ll, ll> pl;
typedef pair<pi, int> pii;
typedef vector<int> vi;
typedef vector<ll> vl;
typedef vector<pi> vii;
typedef vector<pl> vll;
typedef vector<pii> viii;
const int inf = 0x3F3F3F3F;
const ll infl = 0x3F3F3F3F3F3F3F3FLL;
const int mod = 1e9 + 7;
int n; string dir, pre;
int main(){
    //freopen("test.txt", "r", stdin);
    ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);
    for(cin >> n; n != 99999; cin >> n){
        int a = n /10000, b = n/1000, val = n % 1000;
        if((a+b) % 2 != 0) dir = "left";
        else if(a + b == 0) dir = pre;
        else dir = "right";
        pre = dir;
        cout << dir << " " << val << "\n";
    }

}